src/Akka.Hosting/Configuration/EnvironmentVariablesConfigurationHoconAdapter.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -----------------------------------------------------------------------
+//   <copyright file="EnvironmentVariablesConfigurationHoconAdapter.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2023 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//   </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.Extensions.Configuration.EnvironmentVariables;
+
+namespace Akka.Hosting.Configuration;
+
+internal class EnvironmentVariablesConfigurationHoconAdapter: EnvironmentVariablesConfigurationProvider
+{
+    public static EnvironmentVariablesConfigurationHoconAdapter Create(EnvironmentVariablesConfigurationProvider parent)
+    {
+        var propertyInfo = typeof(EnvironmentVariablesConfigurationProvider).GetProperty("Data", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
We have to do some reflection magic here, we can not re-create the `IConfigurationRoot` because some `IConfigurationProvider` can not be run twice (such as the `StreamConfigurationProvider`), so we have to do a raw copy of the existing environment variable provider by reflecting its internal data.